src/FirstSteps/RPG/Heroes/Hero.cs

```diff
@@ -59,20 +57,26 @@ public bool AddItemToBackpack(Item item)
 
         public bool TrySpendCoins(Item item)
         {
-            if (_coinsBag >= item.Price)
+            if (CanHandleSpecialItem(item))
```

Comment:
It's cleaner to write code in that way:
```
if (SomethingWrongHappend)
{
    throw new Exception();
}
DoWhatYouWantedToDo()
```
instead of
```
if (ItIsOk)
{
    DoWhatYouWantedToDo()
}
else
{
    throw new Exception();
}
```
Then you don't need so many nestings. Also when you read the code you see at the begging that some exceptions are thrown for some conditions.